scripts/deploy_jenkins.sh

```diff
@@ -38,6 +38,12 @@ if [ -z "${ENABLE_OAUTH}" ]; then
    ENABLE_OAUTH=false
 fi
 
+if [ -z "${DOCKER_USERNAME}" -o -z "${DOCKER_PASSWORD}" ]; then
+   echo 'DOCKER_USERNAME and DOCKER_PASSWORD should be provided' && exit 1
+fi
+oc secrets new-dockercfg dockerhub --docker-server=docker.io --docker-username=$DOCKER_USERNAME --docker-password=$DOCKER_PASSWORD --docker-email=fh.team.eng-group@redhat.com
```

Comment:
Given that this is an open project that others use than ourselves (and other people do, apparently!), I don't think we should be hardcoding the email here. It's possible that the email parameter is no longer necessary (I think Docker no longer uses it, but I guess depending on your version of `oc`, it might still be required.

If it doesn't need to be specified in the latest version of `oc`, then I'd say remove it, otherwise maybe make it an env var like the others. What do you think?

Reply:
Also note that the format of the secret value has changed in the latest `oc`, to the newer version of the config that Docker uses. This is fine on the latest OpenShift, but doesn't work so well on all older versions (depending on the version of Docker that it uses).

That's not something that needs a change here, just some additional context! :)

Reply:
> Also note that the format of the secret value has changed in the latest oc, to the newer version of the config that Docker uses. This is fine on the latest OpenShift, but doesn't work so well on all older versions (depending on the version of Docker that it uses).
> 
> That's not something that needs a change here, just some additional context! :)

So should I check then version of docker version and based on that call `oc`?

Reply:
No I don't think so (I'm not even sure if that would be possible). We should _soon_ be running only 3.7+, and there even the older format will work, so it won't matter if someone uses an older `oc`. Like I said, it's not something that needs a change here, I just wanted to add it here as extra contextual information.

If someone else is using this, and hits an issue because they're running an older OpenShift version, but have a newer `oc`, then they've got a few options:

- Use a matching version of `oc` for their OpenShift
- Upgrade to 3.7+
- Use an older commit of this script